wolvic/browser/dialogs/user_dialog_manager_bridge.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "wolvic/browser/dialogs/user_dialog_manager_bridge.h"
+
+#include <algorithm>
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "content/public/browser/browser_thread.h"
+#include "wolvic/jni_headers/UserDialogManagerBridge_jni.h"
+
+using base::android::AttachCurrentThread;
+using base::android::ConvertJavaStringToUTF16;
+using base::android::ConvertUTF16ToJavaString;
+using base::android::JavaParamRef;
+using base::android::ScopedJavaGlobalRef;
+
+namespace wolvic {
+
+namespace {
+UserDialogManagerBridge* g_instance = nullptr;
+}
+
+struct InProgressDialog {
+  explicit InProgressDialog(DialogCallback callback);
+  InProgressDialog(const InProgressDialog&) = delete;
+  InProgressDialog& operator=(const InProgressDialog&) = delete;
+  ~InProgressDialog();
+
+  DialogCallback callback;
+};
+
+InProgressDialog::InProgressDialog(DialogCallback callback)
+    : callback(std::move(callback)) {}
+
+InProgressDialog::~InProgressDialog() = default;
+
+UserDialogManagerBridge::UserDialogManagerBridge() {
+  DCHECK(!g_instance);
+  g_instance = this;
```

Comment:
I have a suggestion for an alternative implementation.
1. Remove g_instance
2. Make constructor private and default
3. Make destructor default
4. Implement GetInstance() like
   ```
   static base::NoDestructor<UserDialogManagerBridge> instance;
   return instance.get();
   ```